Jetpack Section: Display a warning if server credentials aren't configured

momo-ozawa opened the pull request:
Fixes #15924

### Description
- Updated the Restore Options VC to display a warning if server credentials aren't configured yet

### How to test

1. Create a Jetpack site
2. Add a Jetpack Security Daily plan
3. Do NOT enable restores (i.e. don't configure your server credentials)
4. Go to My Site > Jetpack > Backup > ellipsis > Restore
5. ✅ Notice that the "Restore to this point" button is disabled
6. ✅ Notice that the restore type switches are disabled and cannot be toggled
7. Tap on the warning "Enter your server credentials..."
8. ✅ Notice that the Jetpack Settings page opens up in a web vc

### Screenshot

Restore Options | Tap on warning to open Jetpack Settings
--- | ---
![Simulator Screen Shot - iPhone 12 Pro - 2021-02-22 at 18 34 17](https://user-images.githubusercontent.com/6711616/108690440-b8dedc00-753d-11eb-9ea5-d05d67d7aa00.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-02-22 at 17 56 49](https://user-images.githubusercontent.com/6711616/108690464-c005ea00-753d-11eb-89c5-14b40fb7fcd1.png)



### PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Update restore options vc to handle awaiting credentials state

Commit:
Add action for detail action button tapped

Commit:
Prevent interaction on switch rows if awaiting credentials

Commit:
Rename detail button to warning button

Commit:
Fix unit tests

emilylaguna commented on WordPress/Classes/Models/JetpackSiteRef.swift:
```diff
@@ -15,6 +15,8 @@ struct JetpackSiteRef: Hashable, Codable {
     let username: String
     /// The homeURL string  for a site.
     let homeURL: String
+    /// The dipslayURL string for a site.
```
```suggestion
    /// The displayURL string for a site.
```

emilylaguna commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Options/JetpackRestoreOptionsViewController.swift:
```diff
@@ -46,4 +56,20 @@ class JetpackRestoreOptionsViewController: BaseRestoreOptionsViewController {
+
+            let title = NSLocalizedString("Unable to visit Jetpack settings for site", comment: "Message displayed when visiting the Jetpack settings page fails.")
+            let notice = Notice(title: title)
+            ActionDispatcher.dispatch(NoticeAction.post(notice))
```
I learned recently you can just do `displayNotice` on a VC, https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Extensions/UIViewController+Notice.swift

```suggestion
            displayNotice(title: title)
```

momo-ozawa replied:
Fixed! 58824fa

emilylaguna commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Options/JetpackRestoreOptionsViewController.swift:
```diff
@@ -1,6 +1,7 @@
 import Foundation
 import CocoaLumberjack
 import Gridicons
+import WordPressFlux
```
See comment below.

```suggestion
```

emilylaguna commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Options/JetpackRestoreOptionsViewController.swift:
```diff
@@ -46,4 +56,20 @@ class JetpackRestoreOptionsViewController: BaseRestoreOptionsViewController {
     }
 
+    override func detailActionButtonTapped() {
+        guard let jetpackSettingsURL = URL(string: "https://wordpress.com/settings/jetpack/\(site.displayURL)") else {
```
When using my testing site that includes a `/` in the URL the WebView ends up white and doesn't load. We can avoid this by passing the site ID instead of the URL. 

```suggestion
        guard let jetpackSettingsURL = URL(string: "https://wordpress.com/settings/jetpack/\(site.siteID)") else {
```

momo-ozawa replied:
Fixed! d1057a0

emilylaguna commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Options/JetpackRestoreOptionsViewController.swift:
```diff
@@ -46,4 +56,20 @@ class JetpackRestoreOptionsViewController: BaseRestoreOptionsViewController {
+        let webVC = WebViewControllerFactory.controller(url: jetpackSettingsURL)
+        let navigationVC = LightNavigationController(rootViewController: webVC)
+
+        self.present(navigationVC, animated: true)
```
```suggestion
       present(navigationVC, animated: true)
```

emilylaguna commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Options/JetpackRestoreOptionsViewController.swift:
```diff
@@ -46,4 +56,20 @@ class JetpackRestoreOptionsViewController: BaseRestoreOptionsViewController {
+        }
+
+        let webVC = WebViewControllerFactory.controller(url: jetpackSettingsURL)
+        let navigationVC = LightNavigationController(rootViewController: webVC)
```
I don't think we need to force the light nav controller here. 

```suggestion
        let navigationVC = UINavigationController(rootViewController: webVC)
```

momo-ozawa replied:
Fixed! abf6b1a

emilylaguna reviewed (changes requested):
Perfect, thanks for tackling this! I left a couple comments below.

jkmassel commented:
👋 We're freezing `16.8` today, so this PR is being bumped to `16.9`. If you need this to be part of the `16.8` release, please merge it into the `release/16.8` branch and DM me – I'll be happy to cut a new beta release!

Commit:
Use displayNotice extension method

Commit:
Update jetpack settings site to use site ID

If a site includes a / in the URL, the WebView ends up white and doesn't load. We can avoid this by passing the site ID instead of the display URL.

Commit:
Use UINavController instead of LightNavController

Commit:
Point WPKit to 4.28-beta

Commit:
Merge branch 'develop' into task/15924-restore-awaiting-credentials

# Conflicts:
#	Podfile
#	Podfile.lock

momo-ozawa commented:
Looks like https://github.com/wordpress-mobile/WordPress-iOS/pull/15958 needs to be merged first.
https://github.com/wordpress-mobile/WordPress-iOS/pull/15958 updates WP with changes introduced in WPKit 4.28.0-beta.1, while this PR uses WPKit 4.28.0-beta.2

momo-ozawa commented:
@emilylaguna Thanks for reviewing! Ready for another look 😄 
Just a note: tests should pass once #15958 is merged

emilylaguna reviewed (approved):
Nice! Once the build issue is fixed. 🚢 🚀

Commit:
Merge branch 'develop' into task/15924-restore-awaiting-credentials

# Conflicts:
#	Podfile.lock

Commit:
Fix typo

momo-ozawa commented:
~Closing this in favor of https://github.com/wordpress-mobile/WordPress-iOS/pull/15981
https://github.com/wordpress-mobile/WordPress-iOS/pull/15981 uses release/16.8 as the base branch~
